doc/nrf/app_dev/device_guides/nrf54l/kmu_provision.rst

```diff
@@ -56,62 +57,84 @@ If the SoC has been previously provisioned and you need to use a different set o
+        .. code-block:: YAML
 
-The script generates the public key for each private key and uploads them to your device.
-These public keys generate the verification keys for the application image, which are then used by MCUboot for validation.
-The first key specified in the command is used for signing the application image.
-Currently, the script supports only ED25519 Keys.
+            - keyname: UROT_PUBKEY
+                keys: ["/path/private-key1.pem", "/path/private-key2.pem"]
+                policy: lock
+            - keyname: APP_PUBKEY
+                keys: ["/path/private-key3.pem", "/path/private-key4.pem"]
+                policy: lock
 
-For MCUboot, take note of the following:
+      * Parameter ``--dry-run`` specifies that a command should generate a keyfile for nRF Util without actually executing the command.
 
-* By default, it uses one key.
-* KMU support in its configuration needs to be enabled by setting the ``SB_CONFIG_MCUBOOT_SIGNATURE_USING_KMU`` sysbuild Kconfig option.
-  Otherwise, MCUboot will fallback to the compiled in key.
+      The script generates the public key for each private key and uploads them to your device.
+      These public keys generate the verification keys for the application image, which are then used by MCUboot for validation.
+      The first key specified in the command is used for signing the application image.
+      Currently, the script supports only ED25519 Keys.
 
-For provision one key to the board run the following command:
+      For MCUboot, take note of the following:
 
-.. parsed-literal::
-   :class: highlight
+      * By default, it uses one key.
+      * KMU support in its configuration needs to be enabled by setting the ``SB_CONFIG_MCUBOOT_SIGNATURE_USING_KMU`` sysbuild Kconfig option.
+        Otherwise, MCUboot will fallback to the compiled in key.
+
+      For provision one key to the board run the following command:
+
+      .. parsed-literal::
+        :class: highlight
+
+          west ncs-provision upload -s nrf54l15 -k ed25519.pem --keyname UROT_PUBKEY
+
+   .. tab:: nRF Util
+
+      Using nRF Util requires generating keys with the `generate_psa_key_attributes.py`_ script, :ref:`similarly to nRF54H20<ug_nrf54h20_keys_generating>`.
+      You need to generate a JSON file with the keys and then provision them to the device using that file.
+      The ``generate_psa_key_attributes.py`` script lets you take keys from a PEM file (using the ``--key-from-file`` option) and generate a JSON file with the keys (using the ``--file`` option).
+
+      To provision keys onto the KMU of the nRF54L15 SoC, use the following nRF Util command, with the ``<snr>`` being the serial number of the device and ``<key-file>`` being the name of the key file in the JSON format:
+
+      .. parsed-literal::
+        :class: highlight
+
+         nrfutil device x-provision-keys --serial-number <snr> --key-file <JSON-key-file>
+
+      You cannot provision several files at once using nRF Util.
```

Comment:
The file can contain multiple keys, you just call generate_psa_key_attributes.py multiple times passing the same file to the --file argument. When you are done adding keys to the .json file you can pass this json file to x-provision-keys and it will provision all of the keys in the json file. You can call x-provision-keys multiple times also to provision multiple keys as long as each key has a different ID which is part of the metadata string. Actually this last requirement apply to both methods of adding multiple keys.